### ISSUE TITLE
No module named 'sslAlgo'

### ISSUE THREAD

**Issue by xhan233** [ZJUJeffLai/SAW_SSL#1]
Hi. Thanks for the great work here.
When I reproduce your work，I meet the fault "No module named 'sslAlgo'",and I wonder if you forget to upload this folder?
Hope to receive your reply

**Comment by ZJUJeffLai** [comment]
Hi, thanks for your reminder, I just uploaded it.

**Comment by xhan233** [comment]
Thanks for your reply,and I have runed the code successfully,but I have another queation about the statistics in your paper:whether the result in Table 1 FixMatch+SAW is weigthed by class_weight_u and class_weight_x simultaneously and what's the type(pseudo or gt)，can you release the sh file?

**Comment by ZJUJeffLai** [comment]
Hi, yes, DISTBL="gt_l" and DISTBU="pseudo", see the updated train_example.sh

**Comment by xhan233** [comment]
Great!And I have reproduced the result.But I am a bit confused about the metric calculation, especially balanced accuracy (bACC): I understand that bACC = (Sensitivity + Specificity) / 2,and when the test set is balanced,the bACC is equal to top-1 accuracy,but they are not euqal when the test set is imbalanced.So where is the code about the metric when the test set is imbalanced(which is your result reported on Table4)?

**Comment by ZJUJeffLai** [comment]
Hi here is our evaluation codes for balanced set: https://github.com/ZJUJeffLai/SAW_SSL/blob/main/train_fix_cifar10.py#L481
And there is for the imbalanced set: 
https://github.com/ZJUJeffLai/SAW_SSL/blob/main/train_fix_cifar10.py#L951

**Comment by xhan233** [comment]
Yes,I know what you mean.It seems that you just report top-1 accuracy when the test set is imbalanced,but not the bACC?

**Comment by ZJUJeffLai** [comment]
Thank you so much for pointing this out! I'll revise our manuscript for this.

**Comment by ZJUJeffLai** [comment]
BTW, would you mind leaving your contact? We can discuss more for such issues if needed.

**Comment by xhan233** [comment]
han_xie@foxmail.com

And I have another confusion,so sorry for bothering you.From the code point of view, When I run the train_example as you provided,I can get the result of Table4(FixMatch+SAW γ=100)from the reverse acc and reverse GM in testLog.txt, but the result I get is much worse than the paper, so how can you get the Table4 result?

**Comment by xhan233** [comment]
> Thank you so much for pointing this out! I'll revise our manuscript for this.

I focus on this just from the issue of DARP


[metric issue](https://github.com/bbuing9/DARP/issues/9)

So when the test set is imbalanced,the bAcc is not equal to top-1 accuracy

**Comment by ZJUJeffLai** [comment]
Hi, actually, the reverse acc and reverse GM in testLog.txt will be the highest number in that column as we only have training/validation settings as previous works, and there is no hold-out test set.

**Comment by xhan233** [comment]
> Hi, actually, the reverse acc and reverse GM in testLog.txt will be the highest number in that column as we only have training/validation settings as previous works, and there is no hold-out test set.

I don't quite understand your mean.You mean we can't use the number in testLog.txt?Then how can I get the Tabel 4(FixMatch+SAW γ=100)result?

**Comment by ZJUJeffLai** [comment]
Hi, I mean, the number is the maximum one for **that column**. May I ask what number you get? 
The number is from testLog.txt, but we choose the maximum one in the column of "reversed bACC" and "reversed GM". Hope this helps.

**Comment by xhan233** [comment]
The maximum number I get from the "reversed bAcc“  is 59.04,and the "reversed GM" is  76.05，so the "reversed Acc"is worse than 64.3 in your paper,is it normal to be that much lower?

**Comment by ZJUJeffLai** [comment]
Hi, I uploaded the raw testLog.txt file we ran last year: 
https://github.com/ZJUJeffLai/SAW_SSL/blob/main/testLog_FixMatch_100_100_cifar10.txt 
The fifth and the sixth columns denote Reversed Acc. and Reversed GM.

**Comment by xhan233** [comment]
> Hi, I uploaded the raw testLog.txt file we ran last year: https://github.com/ZJUJeffLai/SAW_SSL/blob/main/testLog_FixMatch_100_100_cifar10.txt The fifth and the sixth columns denote Reversed Acc. and Reversed GM.

Thanks,you are so kind and patient!

**Comment by ZJUJeffLai** [comment]
Thank you for your interest, actually, I agree with you to some degree that semi-supervised learning field is somehow messy. For example, a lot of works they don't have strict settings as training/validation/test sets, then just report the number of the validation set, which may not be fair to my understanding. Hope we can have a better standard settings here.

**Comment by xhan233** [comment]
> BTW, would you mind leaving your contact? We can discuss more for such issues if needed.

I found that sklearn has a balanced_accuracy method to calculate bacc, hope this helps you！

**Comment by ZJUJeffLai** [comment]
Thank you for sharing this with me, this is super helpful!
